### PR TITLE
Conflict Free Walkthrough Feature

### PR DESCRIPTION
**Pull request by Kilo-Loco**
I added a walkthrough feature as an install option which can be triggered by adding `--walkthrough` or `-w`. It constructs the default install command by asking for the `project`, `targets` and `destination` in a series of questions. Ideally, this series of questions will make it easier for the user to understand what is expected in the install command.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,6 +121,21 @@ $ mockingbird install \
...
 
+Alternatively, you can use the `--walkthrough` or `-w` option to be guided through the install command.
+
+```bash
+$ mockingbird install --walkthrough
+
+Enter an Xcode project.
+Bird.xcodeproj
+
+Which target(s) contain the protocols you want to mock?
```
Comment (andrewchang-bird) [inline README.md]: nit: Mockingbird does both protocols and classes

File: MockingbirdCli/Interface/ArgumentParser+Extensions.swift [hunk 2]
```diff
@@ -160,6 +174,38 @@ extension ArgumentParser {
...
 extension ArgumentParser.Result {
+  func hasWalkthroughOption(using argument: OptionArgument<Bool>) -> Bool {
+    return get(argument) != nil
+  }
+    
+    func getWalkthroughResult(environment: [String: String]) throws -> (project: Path, sources: [String], destination: String) {
+        
+        // Getting project path
+        print("\nEnter an Xcode project.")
+        guard let project = readLine() ?? environment["PROJECT_FILE_PATH"] else {
```
Comment (andrewchang-bird) [inline MockingbirdCli/Interface/ArgumentParser+Extensions.swift]: `0.7.0` now supports project path inference. Perhaps we could pass the inferred value as a default somehow (will require refactoring though).

File: MockingbirdCli/Interface/ArgumentParser+Extensions.swift [hunk 3]
```diff
@@ -160,6 +174,38 @@ extension ArgumentParser {
 }
 
 extension ArgumentParser.Result {
+  func hasWalkthroughOption(using argument: OptionArgument<Bool>) -> Bool {
+    return get(argument) != nil
+  }
+    
+    func getWalkthroughResult(environment: [String: String]) throws -> (project: Path, sources: [String], destination: String) {
```
Comment (andrewchang-bird) [inline MockingbirdCli/Interface/ArgumentParser+Extensions.swift]: nit: At Bird we use two-space indentation. (Adding SwiftLint is on my TODO)

File: MockingbirdCli/Interface/ArgumentParser+Extensions.swift [hunk 4]
```diff
@@ -160,6 +174,38 @@ extension ArgumentParser {
...
+        
+        // Getting project path
+        print("\nEnter an Xcode project.")
+        guard let project = readLine() ?? environment["PROJECT_FILE_PATH"] else {
+            throw ArgumentParserError.expectedValue(option: "<xcodeproj file path>")
+        }
+        let projectPath = Path(project).absolute()
+        guard projectPath.isDirectory, projectPath.extension == "xcodeproj" else {
+            throw ArgumentParserError.expectedValue(option: "--project <xcodeproj file path>")
```
Comment (andrewchang-bird) [inline MockingbirdCli/Interface/ArgumentParser+Extensions.swift]: `--project <xcodeproj file path>` should be `<xcodeproj file path>`

File: MockingbirdCli/Interface/ArgumentParser+Extensions.swift [hunk 5]
```diff
@@ -160,6 +174,38 @@ extension ArgumentParser {
 }
 
 extension ArgumentParser.Result {
+  func hasWalkthroughOption(using argument: OptionArgument<Bool>) -> Bool {
+    return get(argument) != nil
```
Comment (andrewchang-bird) [inline MockingbirdCli/Interface/ArgumentParser+Extensions.swift]: Should probably check that the option is `true` rather than non-nil

**Comment by Kilo-Loco** [comment]
I will make changes based on your suggestions. No problem with changing the name to interactive. This is all still a learning experience in regards to command line apps. Do I close the conversation, or how does this usually work?

**Comment by andrewchang-bird** [comment]
> Do I close the conversation, or how does this usually work?
You can continue adding commits to your branch and the PR will update accordingly.

**Comment by andrewchang-bird** [comment]
Thanks again for the PR. Closing this one out as I think the setup process has improved a bit.

**Comment by chc24** [comment]
nit: spelling `addInteravtiveOption`

**Pull request closed without merging**